core/trino-main/src/main/java/io/trino/execution/TaskId.java

```diff
@@ -35,56 +33,61 @@
     @JsonCreator
     public static TaskId valueOf(String taskId)
     {
-        return new TaskId(taskId);
+        requireNonNull(taskId);
+        var parts = parseDottedId(taskId, 4, "taskId");
+        StageId stageId = StageId.valueOf(parts.subList(0, 2));
+        int partitionId = parseInt(parts.get(2));
+        int attemptId = parseInt(parts.get(3));
+        return new TaskId(stageId, partitionId, attemptId);
     }
 
-    private final String fullId;
+    private final StageId stageId;
+
+    private final int partitionId;
+
+    private final int attemptId;
 
     public TaskId(StageId stageId, int partitionId, int attemptId)
     {
         requireNonNull(stageId, "stageId is null");
         checkArgument(partitionId >= 0, "partitionId is negative");
         checkArgument(attemptId >= 0, "attemptId is negative");
```

Comment:
Add a check like this for `id` field in stage id

Reply:
done.